### ISSUE TITLE
High memory usage

### ISSUE THREAD

**Issue by steve-offutt** [forensicanalysis/artifactcollector#141]
I am collecting the `$MFT` from an NTFS volume on a Windows host using the `goartifact` definition `NTFSMFTFiles`. While doing so I notice a memory usage of around ~400MB. The size of the `$MFT` is about 388MB. This also happens with other larger locked files. I do not observe high memory usage while copying a custom large file. I have produced a big file with `dd if=/dev/urandom of=/tmp/bigfile bs=64M count=32` and it surprisingly does not have high memory usage. However, collecting the `$MFT` does.

VM Host information
```
Edition	Windows 10 Pro
Version	20H2
Installed on	‎2/‎28/‎2021
OS build	19042.1110
Experience	Windows Feature Experience Pack 120.2212.3530.0
```

Example script:
```go
// pretty much customcollector with zip 
package main

import (
	"log"
	"os"

	"github.com/forensicanalysis/artifactcollector/collection"
	"github.com/forensicanalysis/artifactcollector/run"
	"github.com/forensicanalysis/artifactcollector/zipwrite"
	"github.com/forensicanalysis/artifactlib/goartifacts"
	"github.com/forensicanalysis/artifactsgo"
)

func main() {
	f, err := os.Create("collection.zip")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()

	var fs = zipwrite.New(f)
	defer fs.Close()

	bigFile := goartifacts.ArtifactDefinition{
		Name: "bigfile",
		Sources: []goartifacts.Source{
			{
				Type: "FILE",
				Attributes: goartifacts.Attributes{
					Paths: []string{
						"/tmp/bigfile",
						`C:\temp\bigfile`,
					},
				},
				SupportedOs: []string{"Linux", "Windows"},
			},
		},
	}

	arts := append(artifactsgo.Artifacts, bigFile)
	config := collection.Configuration{
		Artifacts: []string{"NTFSMFTFiles", "bigfile"},
		FS:        fs,
	}

	run.Run(&config, arts, nil)
}

```
I've tried to see if artifactcollector could be using the memory. However, the only place I've seen a copy is here:
https://github.com/forensicanalysis/artifactcollector/blob/54c433a6f7ea421d0c22b62819d6a7b5522c2e60/collection/file.go#L182

Is there any other place where artifactcollector may be holding onto bytes after a copy? (I don't believe `collection.hashCopy` to be the culprit.)

**Comment by cugu** [comment]
In the background the artifactcollector uses sqlite to create the forensicstore. It could be sqlite that keeps this data in memory instead of flushing it to disk directly. Is the memory usage an issue for you?

**Comment by steve-offutt** [comment]
Good call on the `crawshaw.io/sqlite` dependency. However, I was able to bypass that dependency with a modified version of `run.Run` that uses the `run.simpleStore` type and I still observe high memory usage while collecting the `$MFT` file with this change. While it is certainly not an issue on most hosts to use 400MB to collect a 388MB file, I could see instances where collecting a much larger file would cause issues. For example, I've observed EVTX files that are >1GB. While rare, this would cause issues if the host does not have enough RAM to perform the copy.
